src/model.ts

```diff
@@ -1084,7 +1184,37 @@ export class VerticalbarColumn extends ValueColumn<number[]> {
     const a_val = this.getValue(a);
     const b_val = this.getValue(b);
 
-    return (d3.sum(a_val) / a_val.length) - (d3.sum(b_val) / b_val.length);
+    const a1_sum = d3.sum(a_val);
+    const a1_min = d3.min(a_val);
+    const a1_max = d3.max(a_val);
+    const a1_mean = a1_sum / a_val.length;
+
+
+    const b1_sum = d3.sum(b_val);
+    const b1_min = d3.min(b_val);
+    const b1_max = d3.max(b_val);
+    const b1_mean = b1_sum / b_val.length;
+
+
+    if (this.desc.sort === 'min') {
+
+
+      return (a1_min) - (b1_min);
+    } else if (this.desc.sort === 'max') {
+
+
+
+      return (a1_max) - (b1_max);
+    } else if (this.desc.sort === 'mean') {
+
+
+      return (a1_mean) - (b1_mean);
+    } else {
+
+      return (d3.sum(a_val)) - (d3.sum(b_val));
+
+    }
+
   }
 }
 export class VerticalconColumn extends ValueColumn<number[]> {
```

Comment:
mulitiple classes contain a lot of code duplication (copy-paste). Which is bad coding style. You could use a base class instead